speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/FileDataStorageTests.cpp

```diff
@@ -107,376 +115,394 @@ DecisionTuple GenerateDecisions(Random& p_random, bool p_doSteer, bool p_doBrake
+TEST_CASE(FileDataStorageTests, DecisionsDisabledNoDecisions, TestWriteDecisions, (false, false, false, false, false, false))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should not be saved but all decisions are made
-TEST_CASE(FileDataStorageTests, DecisionsDisabledAllDecisions, TestDataStorageSaveDecisions, (false, true, true, true, true, true))
+TEST_CASE(FileDataStorageTests, DecisionsDisabledAllDecisions, TestWriteDecisions, (false, true, true, true, true, true))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should be saved; only light decisions are made
-TEST_CASE(FileDataStorageTests, DecisionsLightStored, TestDataStorageSaveDecisions, (true, false, false, false, false, true))
+TEST_CASE(FileDataStorageTests, DecisionsLightStored, TestWriteDecisions, (true, false, false, false, false, true))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should be saved; only gear decisions are made
-TEST_CASE(FileDataStorageTests, DecisionsGearStored, TestDataStorageSaveDecisions, (true, false, false, false, true, false))
+TEST_CASE(FileDataStorageTests, DecisionsGearStored, TestWriteDecisions, (true, false, false, false, true, false))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should be saved; only steer accelerate are made
-TEST_CASE(FileDataStorageTests, DecisionsAccelStored, TestDataStorageSaveDecisions, (true, false, false, true, false, false))
+TEST_CASE(FileDataStorageTests, DecisionsAccelStored, TestWriteDecisions, (true, false, false, true, false, false))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should be saved; only brake decisions are made
-TEST_CASE(FileDataStorageTests, DecisionsBrakeStored, TestDataStorageSaveDecisions, (true, false, true, false, false, false))
+TEST_CASE(FileDataStorageTests, DecisionsBrakeStored, TestWriteDecisions, (true, false, true, false, false, false))
 
 /// @brief Test whether the FileDataStorage stores properly when decisions should be saved; only steer decisions are made
-TEST_CASE(FileDataStorageTests, DecisionsSteerStored, TestDataStorageSaveDecisions, (true, true, false, false, false, false))
+TEST_CASE(FileDataStorageTests, DecisionsSteerStored, TestWriteDecisions, (true, true, false, false, false, false))
 
 /// @brief Test whether the FileDataStorage stores properly for every combination of decision made
-TEST(FileDataStorageTests, DecisionsCombinations)
+TEST(FileDataStorageTests, PairWiseTestWriteDecisions)
 {
     bool alwaysTrue[1]{true};
     bool booleans[2]{true, false};
-    PairWiseTest(TestDataStorageSaveDecisions, alwaysTrue, 1, booleans, 2, booleans, 2, booleans, 2, booleans, 2, booleans, 2);
+    PairWiseTest(TestWriteDecisions, alwaysTrue, 1, booleans, 2, booleans, 2, booleans, 2, booleans, 2, booleans, 2);
```

Comment:
Can be combinatorial as it is only 32 tests and it is better to test all of them